document-store/src/main/java/org/hypertrace/core/documentstore/postgres/update/parser/PostgresIncrementValueParser.java

```diff
@@ -0,0 +1,26 @@
+package org.hypertrace.core.documentstore.postgres.update.parser;
+
+import static org.hypertrace.core.documentstore.postgres.utils.PostgresUtils.formatSubDocPath;
+
+import org.hypertrace.core.documentstore.postgres.Params;
+import org.hypertrace.core.documentstore.postgres.subdoc.PostgresSubDocumentValueParser;
+
+public class PostgresIncrementValueParser implements PostgresUpdateOperationParser {
+  @Override
+  public String parseInternal(UpdateParserInput input) {
+    return new PostgresSetValueParser(this, 1).parseInternal(input);
+  }
+
+  @Override
+  public String parseLeaf(UpdateParserInput input) {
+    final Params.Builder paramsBuilder = input.getParamsBuilder();
+    final PostgresSubDocumentValueParser valueParser =
+        new PostgresSubDocumentValueParser(paramsBuilder);
+
+    paramsBuilder.addObjectParam(formatSubDocPath(input.getPath()[0]));
+    final String parsedValue = input.getUpdate().getSubDocumentValue().accept(valueParser);
+    return String.format(
+        "jsonb_set(%s, ?::text[], (COALESCE(%s->>'%s', '0')::float + %s::float)::text::jsonb)",
```

Comment:
Actually, this seems to be a bigger effort. Let's check the behavior of both databases. If they are the same (either throwing an error or considering a 0 value), let's just add an integration test to verify that.
That way, even if a third database comes in the future, it would be forced to adhere to the same behavior.

In case both Postgres and Mongo are behaving differently, we don't have a way but to add the explicit check.